safe-global/web-core
# [Epic] EIP-1271 support

## What it solves

Resolves #1088

## How this PR fixes it

### `web-core`
- [x] "Messages" routes/tab #1155
- [x] Accordion summary #1156
- [x] Accordion details #1157
- [x] Accordion signature stepper #1158
- [x] Fetching of messages #1159
  - [x] Store slice #1217
  - [x] Integrate endpoint #1243
  - [x] Poll on tag change #1273 
- [x] Add new event bus #1205
- [x] New signing modal #1206
  - [x] Modal #1224
  - [x] Integrate with `communicator` #1270
  - [x] Integrate endpoint #1243
- [x] Extend the test Safe App #1207

### Safe Apps
- [x] Set off-chain signing https://github.com/safe-global/safe-apps-sdk/pull/407
- [x] Create [dApp that uses WC](https://github.com/5afe/eip-1271-dapp)
- [x] Update the WC Safe App https://github.com/safe-global/safe-react-apps/pull/608


## Found issues 

- [x] https://github.com/safe-global/web-core/issues/1435
- [x] https://github.com/safe-global/web-core/issues/1432
- [x] https://github.com/safe-global/web-core/issues/1431
- [x] https://github.com/safe-global/web-core/issues/1430
- [x] https://github.com/safe-global/web-core/issues/1438

## Known issues

- Nested Safes do not receive signatures from the parent

## How to test it

Prerequisites:
- Open the https://github.com/safe-global/safe-react-apps/pull/608 deployment.
- Run the [EIP-1271 test DApp](https://github.com/5afe/eip-1271-dapp) locally.

Method:
1. Open the aforementioned WC app deployment.
2. Connect to the aforementioned test DApp.
3. Toggle off-chain signing and attempt to sign a message.
4. Observe the off-chain signing modal.
5. Sign the message.
6. Observe the signed message in the "Signatures" tab of the transaction list.

## Analytics changes

1. Signing a message dispatches a "Sign message"/"Confirm message" "wallet" event accodingly.
2. Clicking "Sign" dispatches a "Sign message" "tx-list" event.
3. `safe_setSettings` RPC calls should be tracked when enabling/disabling off-chain signing.
